sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -378,11 +381,16 @@ private static boolean initConfigurations(final @NotNull SentryOptions options)
 
   /** Close the SDK */
   public static synchronized void close() {
-    final IHub hub = getCurrentHub();
     mainHub = NoOpHub.getInstance();
-    // remove thread local to avoid memory leak
-    currentHub.remove();
-    hub.close();
+    // close the current threads hub
+    Thread currentThread = Thread.currentThread();
+    IHub hub = hubMap.get(currentThread.getId());
+    if (hub != null) {
+      hub.close();
+      // remove the current threads hub from the map
+      hubMap.remove(currentThread.getId());
+    }
```

Comment:
In case we come back to this PR to give it a try in the next major, we should also clear the map on `close`.

Reply:
Sorry for being so slow on this. I have not run this in production yet. I'm a bit sure about the best way to go about testing this. Do you have any tips for me as to how i could test it, to be enough to be included in a future release? We are still dealing with this issue in production, but are just managing with manual restarts of tomcat now and then.

Reply:
We're in the middle of rewriting some parts of the SDK for the upcoming `8.x` release so the PR won't be mergable as is. It should be rather straight forward to adapt it to the new `IScopesStorage` interface.

> Do you have any tips for me as to how i could test it

- Basically spin up the application before and after the change and compare what lands in Sentry.
	- Keep an eye out for data that's missing
	- Also check for data that doesn't belong to errors/transactions/... you're comparing as they might indicate leaks
- Spin up the server locally, hit it with lots of requests, run GC a couple times then capture a thread dump
	- You can compare the results of before and after the change
	- There shouldn't be any additional objects being held on to (`Hub`, `Scope`, `Attachment`, `Breadcrumb`, tags etc.)
- Check whether your tomcat warnings go away when using the updated version

As this affects all of the SDK there is a lot more testing required that's why we haven't yet given this PR the love it deserves.

In `8.x` of the Java SDK we'll have to retest whether this issue even exists anymore since we're introducing lifecycle tokens that you can call `close` on when you're done so maybe that already takes care of cleaning up `ThreadLocals`. I haven't gotten around to testing this yet.